modules/database-commons/src/main/java/org/testcontainers/ext/ScriptUtils.java

```diff
@@ -229,6 +230,18 @@ private static boolean containsSubstringAtOffset(String lowercaseString, String
 		return lowercaseString.startsWith(lowercaseSubstring, offset);
 	}
 
+    private static boolean containsKeywordsAtOffset(String lowercaseString, String keywords, int offset) {
+        String lowercaseSubstring = keywords.toLowerCase();
+
+        boolean isSeperated = offset == 0;
+        if (offset > 0) {
+            char seperator = lowercaseString.charAt(offset - 1);
+            isSeperated = seperator == '\n' || seperator == '\t' || seperator == ' ';
+        }
```

Comment:
Hmm, what about cases where a word begins with the token?

e.g. `CREATE TABLE endings (...)`

I think we need to look after the keyword as well...

Reply:
Yes, you are right. I think we need to check the margin char on both sides of the keywords. One way is to check whether the margin char is a valid name qualifier. It has less impact to the original code.

Reply:
> One way is to check whether the margin char is a valid name qualifier. 

🤔 are you sure about this? Why not check that the character after is not a separator character, the same way that we check the character before?

Otherwise `endings` is OK, but `end2` is equivalent to `end`.

Another thing to be wary of is that:
* `;` is a separator we should look for
* end-of-file should also be treated like a separator that can terminate keywords.

Reply:
I updated the code and do check the separator. However, I still think checking the separator is a little more complicate, it involves more variables and scenarios.  
EOF is not a char so I don't think I need to check it.

Reply:
Thanks @DuanYuePeng, if you don't mind I'll check this out and add more unit tests. Our test coverage has never been good (hence the bug!) and I feel guilty about this.

Reply:
You’re welcome. It’s been a pleasure to work with you on this.